File: 695-max-area-of-island/695-max-area-of-island.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        visit = []
        m = len(grid)
        n = len(grid[0])
        for i in range(m):
            visit.append([False]*n)
        val = 0
        def dfs(grid, visit, r, c):
            if r>=0 and c>=0 and r< len(grid) and c<len(grid[0]):
                if not visit[r][c] and grid[r][c]==1:
                    visit[r][c]=True
                    arr.extend([1])
                    dfs(grid, visit, r+1, c)
                    dfs(grid, visit, r-1, c)
                    dfs(grid, visit, r, c+1)
                    dfs(grid, visit, r, c-1)                       
        for i in range(m):
            for j in range(n):
                arr = []
                if not visit[i][j]:
                    dfs(grid, visit, i, j)
                    if sum(arr)>val:
                        val = sum(arr)
        return val

File: 695-max-area-of-island/695-max-area-of-island.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int max = 0;
        for(int i=0;i<grid.length;i++){
            for(int j=0;j<grid[0].length;j++){
                if(grid[i][j]==1){
                    max = Math.max(max,dfs(grid,i,j));
                }
            }
        }
        return max;
    }
    int dfs(int[][] mat,int i, int j){
        if(i<0 || j<0 || i>=mat.length || j>=mat[0].length || mat[i][j]!=1){
            return 0;
        }
        mat[i][j]=0;
        return 1+dfs(mat,i+1,j)+dfs(mat,i-1,j)+dfs(mat,i,j+1)+dfs(mat,i,j-1);
    }
}